### DISCUSSION TITLE
Extension that adds `integral` functionality to compute simple integrals

### DISCUSSION THREAD

**Discussion by joelahoover** [josdejong/mathjs#2748] [Ideas]
Hi all! I'm pleased to announce the first public release of my current project: [`mathjs-simple-integral`](https://github.com/joelhoover/mathjs-simple-integral) (also [published to npm](https://www.npmjs.com/package/mathjs-simple-integral)). This package implements the `integral` function for Math.js. It is currently in beta and has some rather severe limitations, such as no implementing integration-by-parts or u-substitution, but it is complete enough to be useful in some circumstances (for example, it has full support for polynomial expressions).

A quick overview of the integration algorithm: it uses depth-first search memoized pattern matching that transforms the integrand until it can be integrated directly. The rules for integration are simply an array of functions, with each one matching an integrand pattern. These rules can be specified for each call to `integral` (just like providing custom simplification rules for `simplify`). Head over to the [project's README](https://github.com/joelhoover/mathjs-simple-integral#simple-integral-for-mathjs) for a more in-depth explanation and examples.

As this is a much requested feature for Math.js (#35, #442, #788, #993), I though it would be relevant to post about it here. I would also be willing to let the Math.js team take control over the package and integrate it (pun intended) into Math.js itself, if they are interested.

**Comment by josdejong** [comment]
That looks like an impressive start Joel, awesome! Would be great if we could integrate it (pun intended) in math.js as soon as you feel like it's complete enough to remove the beta label :). I will be happy to give feedback where needed but I'm afraid I lack spare time to help with the development itself.

**Comment by husayt** [comment]
@joelhoover @josdejong this is great stuff indeed. Would be really amazing to closely "integrate" with math.js as suggested.

**Comment by nycos62** [comment]
Hello my friends, any news on integrals ?
I dream of evaluating [Stieltjes constants](https://en.wikipedia.org/wiki/Stieltjes_constants) with math.js 
![formula_((-1)^n_n!)_2pi_integrate(x,0,2_pi,e^(-n_i_x)_zeta(e^(i_x)+1))](https://github.com/user-attachments/assets/66cad49c-34ac-41a4-9c4a-fa2b4200a3b6)
any chance to integrate `e^(-n*i*x)*zeta(e^(i*x)+1) from 0 to 2pi` with math.js soon ?

**Reply by josdejong** [reply]
No news there. Help implementing integrals would be welcome.
